include/NeoFOAM/fields/comparisons/FieldComparison.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2023 NeoFOAM authors
+#pragma once
+
+#include "NeoFOAM/fields/Field.hpp"
+#include <span>
+
+namespace NeoFOAM
+{
+
+template<typename T>
+bool compare(Field<T>& field, T value)
```

Comment:
I think that should be renamed, since it compares for equality. Otherwise we should have a compare function which takes a predicate.
```suggestion
bool equal(Field<T>& field, T value)
```
Also the Field needs to be marked const